### PR TITLE
HyperDX console instrumentation (refactoring) + http instrumentation (advanced network capture)

### PR DESCRIPTION
**Pull request by wrn14897**

### CODE REVIEW

File: packages/node-opentelemetry/src/patch.ts [hunk 1]
```diff
@@ -1,92 +0,0 @@
-import _ from 'lodash';
```
Comment (wrn14897) [inline packages/node-opentelemetry/src/patch.ts:1]: Move this to `otel.ts`

File: packages/node-opentelemetry/src/tracing.ts [hunk 2]
```diff
@@ -1,90 +1,27 @@
-import { NodeSDK } from '@opentelemetry/sdk-node';
```
Comment (wrn14897) [inline packages/node-opentelemetry/src/tracing.ts:1]: Move this to `otel.ts`

File: packages/node-opentelemetry/src/instrumentations.ts [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+  return firstJson
+    ? {
+        ...firstJson,
+        // FIXME: we probably don't want to override 'message' field in firstJson
+        ...(args.length > 1 && { message: stringifiedArgs.join(' ') }),
+      }
+    : stringifiedArgs.join(' ');
+};
+
+const ALL_HTTP_REQUEST_HEADERS = [
```
Comment (MikeShi42) [inline packages/node-opentelemetry/src/instrumentations.ts:45]: I'm concerned this only really captures an arbitrary(?) subset of headers, wondering if we should just bring the [otel header capture implementation](https://github.com/open-telemetry/opentelemetry-js/blob/902229afd89a2cf59b120b75892a56bdab5ff039/experimental/packages/opentelemetry-instrumentation-http/src/http.ts#L345) directly into our req/res hooks if possible?

If not possible, we should probably at least document this default behavior, and give a configuration option so anyone can pass their own custom headers.

File: packages/node-opentelemetry/README.md [hunk 4]
```diff
@@ -90,3 +90,37 @@ Run your application with the following command (example using `ts-node`):
...
+});
+
+// Other instrumentation code...
+```
+
+### Advanced Instrumentation Configuration (Optional)
+
+#### Capture Console Logs
+
+By default, the OpenTelemetry SDK will capture console logs. You can disable it by setting `HDX_JS_CONSOLE_INSTRUMENT` flag to 0.
```
[the remark this answers is not in the document]
Comment (wrn14897) [reply packages/node-opentelemetry/README.md:113]: good point. agreed _NODE_ is better

File: packages/node-opentelemetry/README.md [hunk 5]
```diff
@@ -90,3 +90,37 @@ Run your application with the following command (example using `ts-node`):
...
+
+#### Option 3
+
+You can also initialize the OpenTelemetry SDK manually in your own otel configuration file.
+
+```ts
+import { initSDK } from '@hyperdx/node-opentelemetry';
+
+initSDK({
+  captureConsole: true, // optional, default: true
```
[the remark this answers is not in the document]
Comment (wrn14897) [reply packages/node-opentelemetry/README.md:102]: I tried to align these two env vars with the browser pkg + python sdk. but no big deal, let me rename 'captureConsole'

[the remark this answers is not in the document]
Comment (MikeShi42) [reply packages/node-opentelemetry/README.md:102]: ahhh.... 😅  fun

File: packages/node-opentelemetry/README.md [hunk 6]
```diff
@@ -90,3 +90,37 @@ Run your application with the following command (example using `ts-node`):
...
+
+By default, the OpenTelemetry SDK will capture console logs. You can disable it by setting `HDX_JS_CONSOLE_INSTRUMENT` flag to 0.
+
+```sh
+export HDX_JS_CONSOLE_INSTRUMENT=0
+```
+
+#### Advanced Network Capture
+
+By enabling network capture features, developers gain the capability to debug HTTP request headers and body payloads effectively.
```
[the remark this answers is not in the document]
Comment (wrn14897) [reply packages/node-opentelemetry/README.md:122]: I think currently the instrumentation only works for inbound

File: packages/node-opentelemetry/dummy.js [hunk 7]
```diff
@@ -21,6 +21,7 @@ const logger = winston.createLogger({
 const pinoLogger = pino();
 
 app.get('/', (req, res) => {
+  console.info('@@@@@@@@@@@@');
```
[the remark this answers is not in the document]
Comment (wrn14897) [reply packages/node-opentelemetry/dummy.js:24]: yap yap, not important. just local testing

File: packages/node-opentelemetry/README.md [hunk 8]
```diff
@@ -90,3 +90,37 @@ Run your application with the following command (example using `ts-node`):
...
+```ts
+import { initSDK } from '@hyperdx/node-opentelemetry';
+
+initSDK({
+  captureConsole: true, // optional, default: true
+  advancedNetworkCapture: true, // optional, default: false
+});
+
+// Other instrumentation code...
+```
```
[the remark this answers is not in the document]
Comment (wrn14897) [reply packages/node-opentelemetry/README.md:114]: Updated the doc with link and more details

File: packages/node-opentelemetry/src/instrumentations.ts (hunk 3 above)
Comment (wrn14897) [reply packages/node-opentelemetry/src/instrumentations.ts]: Pushed the update. Now all headers should be captured

**Review by MikeShi42 — APPROVED** [review]

**Pull request merged**